### ISSUE TITLE
Option to use flash_attention

### ISSUE THREAD

**Issue by Sindre01** [uiuc-focal-lab/syncode#182]
It would be very usefull to have the option to use flash attention to increase speed and lower memory usage.

**Comment by shubhamugare** [comment]
If you use SynCode as logits processor you should be able to use flash attention even in the current version. See this [example](https://github.com/structuredllm/syncode/blob/main/notebooks/example_logits_processor.ipynb)
